awd/config.py

```diff
@@ -1,39 +1,45 @@
 import logging
 import os
 
-ENV = os.getenv("WORKSPACE")
-
 logger = logging.getLogger(__name__)
-logger.debug("Configuring awd for current env: %s", ENV)
 
 AWS_REGION_NAME = "us-east-1"
 DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
 
-if ENV == "test":
-    LOG_LEVEL: str | None = "INFO"
-    DOI_TABLE: str | None = "wiley-test"
-    METADATA_URL: str | None = "http://example.com/works/"
-    CONTENT_URL: str | None = "http://example.com/doi/"
-    BUCKET: str | None = "awd"
-    SQS_BASE_URL: str | None = "https://queue.amazonaws.com/123456789012/"
-    SQS_INPUT_QUEUE: str | None = "mock-input-queue"
-    SQS_OUTPUT_QUEUE: str | None = "mock-output-queue"
-    COLLECTION_HANDLE: str | None = "123.4/5678"
-    LOG_SOURCE_EMAIL: str | None = "noreply@example.com"
-    LOG_RECIPIENT_EMAIL: str | None = "mock@mock.mock"
-    RETRY_THRESHOLD: str | None = "10"
-    SENTRY_DSN: str | None = None
-else:
-    LOG_LEVEL = os.getenv("LOG_LEVEL")
-    DOI_TABLE = os.getenv("DOI_TABLE")
-    METADATA_URL = os.getenv("METADATA_URL")
-    CONTENT_URL = os.getenv("CONTENT_URL")
-    BUCKET = os.getenv("BUCKET")
-    SQS_BASE_URL = os.getenv("SQS_BASE_URL")
-    SQS_INPUT_QUEUE = os.getenv("SQS_INPUT_QUEUE")
-    SQS_OUTPUT_QUEUE = os.getenv("SQS_OUTPUT_QUEUE")
-    COLLECTION_HANDLE = os.getenv("COLLECTION_HANDLE")
-    LOG_SOURCE_EMAIL = os.getenv("LOG_SOURCE_EMAIL")
-    LOG_RECIPIENT_EMAIL = os.getenv("LOG_RECIPIENT_EMAIL")
-    RETRY_THRESHOLD = os.getenv("RETRY_THRESHOLD")
-    SENTRY_DSN = os.getenv("SENTRY_DSN")
+config_variables = [
+    "WORKSPACE",
+    "LOG_LEVEL",
+    "DOI_TABLE",
+    "METADATA_URL",
+    "CONTENT_URL",
+    "BUCKET",
+    "SQS_BASE_URL",
+    "SQS_INPUT_QUEUE",
+    "SQS_OUTPUT_QUEUE",
+    "COLLECTION_HANDLE",
+    "LOG_SOURCE_EMAIL",
+    "LOG_RECIPIENT_EMAIL",
+    "RETRY_THRESHOLD",
+    "SENTRY_DSN",
+]
+
+
+class Config:
+    def __init__(self) -> None:
+        self.load_config_variables()
```

Comment:
What about a slight tweak that might address two issue?

If we move the list of require env vars into the `Config` class, and then also define empty attributes that will get populated, this will:

  1. colocate those required env vars on the class itself (allowing for something like `Config.REQUIRED_ENV_VARS` if ever needed)
  
  2. can remove `[attr-defined]` linting skip wherever used for config variables

Something along the lines of:
```python
class Config:
    REQUIRED_ENV_VARS: Iterable[str] = (  # making a tuple + Iterable[str] avoids mutable type which throws linting errors
        "WORKSPACE",
        "LOG_LEVEL",
        "DOI_TABLE",
        "METADATA_URL",
        "CONTENT_URL",
        "BUCKET",
        "SQS_BASE_URL",
        "SQS_INPUT_QUEUE",
        "SQS_OUTPUT_QUEUE",
        "COLLECTION_HANDLE",
        "LOG_SOURCE_EMAIL",
        "LOG_RECIPIENT_EMAIL",
        "RETRY_THRESHOLD",
        "SENTRY_DSN",
    ]

    WORKSPACE: str
    LOG_LEVEL: str
    DOI_TABLE: str
    METADATA_URL: str
    CONTENT_URL: str
    BUCKET: str
    SQS_BASE_URL: str
    SQS_INPUT_QUEUE: str
    SQS_OUTPUT_QUEUE: str
    COLLECTION_HANDLE: str
    LOG_SOURCE_EMAIL: str
    LOG_RECIPIENT_EMAIL: str
    RETRY_THRESHOLD: str
    SENTRY_DSN: str

    def __init__(self) -> None:
        self.load_environment_variables()
    ...
    ....
```

It may feel a bit repetitive, but that feels okay given the conveniences it affords.

An alternative to including the class attribute list `REQUIRED_ENV_VARS`, could be _keeping_ the attributes (e.g. `WORKSPACE: str`), and then looping through `self.__class__.__annotations__` which will provide those attributes as an `<ATTRIBUTE_NAME>:<TYPE>` dictionary, e.g.:

```python
{
 'WORKSPACE': str,
 'LOG_LEVEL': str,
 'DOI_TABLE': str,
 'METADATA_URL': str,
 'CONTENT_URL': str,
 'BUCKET': str,
 'SQS_BASE_URL': str,
 'SQS_INPUT_QUEUE': str,
 'SQS_OUTPUT_QUEUE': str,
 'COLLECTION_HANDLE': str,
 'LOG_SOURCE_EMAIL': str,
 'LOG_RECIPIENT_EMAIL': str,
 'RETRY_THRESHOLD': str,
 'SENTRY_DSN': str}
```

I like this as well, but it's probably unncessary complexity, and runs the risk of any _other_ attributes that might get added getting incorrectly interpreted as env vars to look for.

Reply:
I agree with the 1st approach, the `#type: ignore`'s never feel good and there could be cases where we need to access `config.REQUIRED_ENV_VARS`. I did try this originally and balked at the repetition but on second thought, I think that's better than `#type: ignore`'s. And I think you're right about unnecessary complexity in looping through `self.__class__.__annotations__`

Reply:
I balked at the repitition too... but I think it's okay.  

Perhaps they can be looked at as having slightly different purposes:
  * the first is the list of required env vars, unchanging, authoratative
  * the attributes are typed, provide a mechanism for defaults

If those config attributes grew -- and I think it's possible beyond just env vars -- than those two lists would deviate and it'd desirable.  So I think it's okay!